sys/net/wot/core/serialization/serialization.c

```diff
@@ -1100,8 +1102,11 @@ int wot_td_serialize_thing(wot_td_serialize_receiver_t receiver, wot_td_thing_t
 
     if (thing->context != NULL) {
         _wot_io_send(receiver, "[", 1, slicer);
-        _wot_io_send_string(receiver, _wot_td_url, sizeof(_wot_td_url) - 1, slicer);
-        _wot_io_send(receiver, ",", 1, slicer);
+        if(strcmp(thing->context->value, _wot_td_url))
+        {
```

Comment:
Yes and it should be included by default. Even if you use the library without the generation, there shouldn't be the need to add this by yourself. It is necessary to use WoT. Also it makes it possible for us to control the version we support. We can set this to another version when we are sure that we support all the features set in that version. Or even potentially needed to even support that version.

Reply:
I inserted this line, because if you set the default context in the thing description that gets processed explicitly, the default context get set anyway and the context gets duplicated. In this way the default context should only be set if no context value is given.